frontend/src/scenes/onboarding/Onboarding.tsx

```diff
@@ -132,6 +139,26 @@ const OnboardingWrapper = ({
     return currentOnboardingStep
 }
 
+const sessionReplayOnboardingToggle = (
+    currentTeam: TeamType | TeamPublicType | null,
+    selectedProducts: ProductKey[]
+): ProductConfigOption => {
+    const userDecision =
+        currentTeam?.session_recording_opt_in ||
+        selectedProducts.includes(ProductKey.SESSION_REPLAY) ||
+        currentTeam?.product_intents?.some((intent) => intent.product_type === ProductKey.SESSION_REPLAY)
```

Comment:
potentially we could get away with only checking currentTeam?.productIntents here but...